src/timer.rs

```diff
@@ -0,0 +1,75 @@
+use crate::pac::TIMER0;
+use crate::sysctl::{self, APB0};
+use crate::time::Hertz;
+
+macro_rules! timer {
+    ($Timer: ident, $timer: ident, $TIMER: ident) => {
+        pub struct $Timer<'a> {
+            timer: &'a $TIMER,
+            channel_id: usize,
```

Comment:
I think this should take a `CHANNEL` instance and implement a `free` function to return it back. However, I did not find a way to move a `CHANNEL` out of `timer0::RegisterBlock::channel`